File: backend/src/main/java/application/entities/OrderProducts.java
package application.entities;

import com.fasterxml.jackson.annotation.JsonBackReference;
import com.fasterxml.jackson.annotation.JsonManagedReference;
import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import lombok.Data;
import static jakarta.persistence.FetchType.EAGER;

@Data
@Entity
@Table(name = "orderproducts")
public class OrderProducts {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false)
    private Long id;


    @ManyToOne(fetch = EAGER, optional = false)
    @JoinColumn(name = "order_id")
    @JsonBackReference(value = "orderProducts")
    private Order order;


    @ManyToOne(fetch = EAGER, optional = false)
    @JoinColumn(name = "product_id")
    @JsonBackReference(value = "product")
    private Product product;

    @Basic
    @NotNull
    @Column(name = "quantity")
    private Integer quantity;

    @Basic
    @NotNull
    @Column(name = "price")
    private Double price;

}

File: frontend/src/app/models/Orderproducts.ts
import { Product } from "./Product";
import { Order } from "./Order";
export interface OrderProducts {
  id: number;
  orderId: number;
  productId: number;
  quantity: number;
  price: number;
}
export class OrderProductsImpl implements OrderProducts {
  public id: number;
  public orderId: number;
  public productId: number;
  public quantity: number;
  public price: number;

  constructor(order:Order, product:Product,quantity:number){
    this.id=1;
    this.orderId=order.id;
    this.productId=product.id;
    this.quantity=quantity;
    this.price=product.price;
  }

  copy(op:OrderProducts) : OrderProductsImpl{
    this.id=op.id;
    this.orderId=op.orderId;
    this.productId=op.productId;
    this.price=op.price;
    this.quantity=op.quantity;
    return this;
  }

}
